pkg/karmadactl/addons/init/enable_option.go

```diff
@@ -0,0 +1,157 @@
+package init
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/klog/v2"
+	aggregator "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+	"k8s.io/utils/strings/slices"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	karmadaclientset "github.com/karmada-io/karmada/pkg/generated/clientset/versioned"
+	"github.com/karmada-io/karmada/pkg/karmadactl/cmdinit/utils"
+	"github.com/karmada-io/karmada/pkg/util/gclient"
+)
+
+// CommandAddonsEnableOption options for addons list.
+type CommandAddonsEnableOption struct {
+	// KubeConfig holds host cluster KUBECONFIG file path.
+	KubeConfig string
+	Context    string
+
+	// KubeConfig holds karmada control plane KUBECONFIG file path.
+	KarmadaConfig  string
+	KarmadaContext string
+
+	// Namespace holds the namespace where Karmada components intalled
+	Namespace string
+
+	KarmadaSearchImage string
+
+	KarmadaSearchReplicas int32
+
+	KarmadaDeschedulerImage string
+
+	KarmadaDeschedulerReplicas int32
+
+	KarmadaEstimatorImage string
+
+	KarmadaEstimatorReplicas int32
+
+	KubeClientSet *kubernetes.Clientset
+
+	KarmadaKubeClientSet *kubernetes.Clientset
+
+	KarmadaClientSet *karmadaclientset.Clientset
+
+	KarmadaAggregatorClientSet *aggregator.Clientset
+
+	ControlPlaneClient client.Client
+
+	WaitPodReadyTimeout int
+
+	WaitAPIServiceReadyTimeout int
+
+	Cluster string
+
+	MemberConfig string
+
+	MemberContext string
+}
+
+// Complete the conditions required to be able to run enable.
+func (o *CommandAddonsEnableOption) Complete() error {
+	if o.KubeConfig == "" {
+		env := os.Getenv("KUBECONFIG")
+		if env != "" {
+			o.KubeConfig = env
+		} else {
+			o.KubeConfig = filepath.Join(homedir.HomeDir(), ".kube", "config")
+		}
+	}
+
+	restConfig, err := utils.RestConfig(o.Context, o.KubeConfig)
+	if err != nil {
+		return err
+	}
+
+	o.KubeClientSet, err = utils.NewClientSet(restConfig)
+	if err != nil {
+		return err
+	}
+
+	karmadaRestConfig, err := utils.RestConfig(o.KarmadaContext, o.KarmadaConfig)
+	if err != nil {
+		return err
+	}
+
+	o.KarmadaKubeClientSet, err = utils.NewClientSet(karmadaRestConfig)
+	if err != nil {
+		return err
+	}
+
+	o.KarmadaClientSet, err = utils.NewKarmadaClientSet(karmadaRestConfig)
+	if err != nil {
+		return err
+	}
+
+	o.KarmadaAggregatorClientSet, err = utils.NewAPIRegistrationClient(karmadaRestConfig)
+	if err != nil {
+		return err
+	}
+
+	o.ControlPlaneClient = gclient.NewForConfigOrDie(karmadaRestConfig)
+
+	return nil
+}
+
+// Validate Check that there are enough conditions to run the enable.
+func (o *CommandAddonsEnableOption) Validate(args []string) error {
+	secretClient := o.KubeClientSet.CoreV1().Secrets(o.Namespace)
+	_, err := secretClient.Get(context.TODO(), "kubeconfig", metav1.GetOptions{})
```

Comment:
"kubeconfig" is better to depend on `karmadactl init` code. If we change `karmadactl init`, it can be found fastly, avoid to forget to change it.